lib/presentation/widgets/list_characters.dart

```diff
@@ -1,56 +1,139 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:marvelapp_flutter/presentation/features/home/bloc/home_bloc.dart';
+import 'package:marvelapp_flutter/presentation/features/home/bloc/home_event.dart';
+import 'package:marvelapp_flutter/presentation/features/home/bloc/home_state.dart';
 import 'package:marvelapp_flutter/presentation/models/character_view_data.dart';
 import 'package:marvelapp_flutter/presentation/navigation/app_routes.dart';
+import 'package:marvelapp_flutter/presentation/widgets/bottom_error.dart';
+import 'package:marvelapp_flutter/presentation/widgets/bottom_loader.dart';
+import 'package:marvelapp_flutter/presentation/widgets/empty_widget.dart';
+import 'package:marvelapp_flutter/presentation/widgets/page_error.dart';
 
-class ListCharacters extends StatelessWidget {
-  final List<CharacterViewData>? list;
+import 'center_loader.dart';
 
-  const ListCharacters({Key? key, required this.list}) : super(key: key);
+class ListCharacters extends StatefulWidget {
+  const ListCharacters({Key? key}) : super(key: key);
+
+  @override
+  State<ListCharacters> createState() => _ListCharactersState();
+}
+
+class _ListCharactersState extends State<ListCharacters> {
+  final _scrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(_onScroll);
+  }
 
   @override
   Widget build(BuildContext context) {
+    return BlocBuilder<HomeBloc, HomeState>(
+      builder: (context, state) {
+        Widget listViewBuilder = _getListViewBuilder(state.characters, state.hasReachedMax, state.loading, state.error);
```

Comment:
Looks like a good candidate to exctract to separate widget instead of the private method and it should not contains `loading` parameter. loading should be handled above and the LoadingIndicator shown instead of the heroes list.